### DISCUSSION TITLE
stop match at space

### DISCUSSION THREAD

**Discussion by bjlockie** [moisseev/received#9] [Q&A]
How do I stop matching at the first space?

For example if I have these headers:

`Received: (qmail 576 invoked by uid 108); 22 Jan 2023 16:59:17 +0000
Received: by simscan 1.4.0 ppid: 530, pid: 570, t: 0.0347s
         scanners: clamav: 0.105.0/m:62/d:26789
Received: from unknown (HELO smtp.github.com) (192.30.252.205)
  by example.net with ESMTPS; 22 Jan 2023 16:59:17 +0000
Received: from github-lowworker-d74de0d.ac4-iad.github.net (github-lowworker-d74de0d.ac4-iad.github.net [10.52.100.99])
	by smtp.github.com (Postfix) with ESMTP id CD54272054A
	for <xxx@yyy>; Sun, 22 Jan 2023 08:59:12 -0800 (PST)
`

I TRIED: " ^(from .+) (by .+)(;.+)$" but I just want to show the "from" "by" and date/time.```
but I get:
`from unknown (HELO smtp.github.com) (192.30.252.205) by example.net with ESMTPS; 22 Jan 2023 16:59:17 +0000from github-lowworker-d74de0d.ac4-iad.github.net (github-lowworker-d74de0d.ac4-iad.github.net [10.52.100.99])by smtp.github.com (Postfix) with ESMTP id CD54272054A for <xxx@yyy>; Sun, 22 Jan 2023 08:59:12 -0800 (PST)`

I want (line feeds between each line):
```
from unknown by example.net; 22 Jan 2023 16:59:17 +0000
from github-lowworker-d74de0d.ac4-iad.github.net by smtp.github.com; Sun, 22 Jan 2023 08:59:12 -0800 (PST)
```

Why isn't the date format standard in headers?
It seems like that should have been one of the first fixes to email.

**Comment by moisseev** [comment]
Each Received header should be displayed on separate line. For now it works properly on TB91. I'll fix it for later versions.

Only the characters that match capturing groups are displayed. This means everything you want to hide should match regexp parts that NOT enclosed in braces. In your example it is a single space followed by "by ".

For the same reason `^(from .+) (by .+)(;.+)$` displays the same text as `^(from .+) (by .+;.+)$`. Using two capturing groups doesn't make sense here.

You need regexp like `^from ([^(]+?) \(.+\)( by [^(]+)(?: \(.+)? with.+(; .+?)$`. It's just an example, I'm not sure it covers all corner cases.

You can play with regexps at https://regex101.com/

**Comment by bjlockie** [comment]
I have TB-102.4.2.

Thanks.

What "flavor" of regex should I use on that site?

**Comment by bjlockie** [comment]
This does what I want:
(from \S*).*\s*( by \S*)\s*[^;]*;\s*( .*)

The results are all on one line but that'll get fixed eventually.

Thanks for making such an awesome addon.

**Reply by moisseev** [reply]
Fixed in [2.4.3](https://github.com/moisseev/received/releases/tag/2.4.3)
